config/test.exs

```diff
@@ -0,0 +1,3 @@
+import Config
+
+config :tesla, adapter: {Tesla.Adapter.Hackney, [path_encode_fun: &URI.encode/1]}
```

Comment:
The `path_encode_fun` is brought over from the direct usage of Hackney. Without it, the URL for searching multiple indices has the `*` character get URI-encoded, which Algolia then rejects.

Since this option is adapter-specific, I'm not sure a good way to encode this rule in the library itself in a way that doesn't break with other adapters.